tests/validate/aie_pl_test/src/host.cpp

```diff
@@ -198,13 +134,203 @@ main(int argc, char* argv[])
     out_bo1.sync(XCL_BO_SYNC_BO_FROM_DEVICE, mem_size, /*OFFSET=*/0);
 
     // post-processing data;
-    bool match = 0;
+    bool match = false;
     for (unsigned int i = 0; i < mem_size / sizeof(int); i++) {
         if (*(host_out1 + i) != *(host_in1 + i) + 1) {
-            match = 1;
+            match = true;
             std::cout << boost::format("host_out1[%u]=%d\n") % i % host_out1[i];
         }
     }
+    return match;
+}
+
+bool run_pl_controller_aie2(xrt::device device, xrt::uuid uuid, std::string aie_control, std::string dma_lock) {
+    // instance of plController
+    xf::plctrl::plController_aie2 m_pl_ctrl(aie_control.c_str(), dma_lock.c_str());
+
+    int num_iter = 1;
+    int num_sample = 32;
+
+    m_pl_ctrl.enqueue_set_aie_iteration("mygraph", num_iter);
+    m_pl_ctrl.enqueue_enable_aie_cores();
+
+    for (int i = 0; i < num_iter; ++i) {
+        m_pl_ctrl.enqueue_sync();
+    }
+
+    m_pl_ctrl.enqueue_sleep(128);
+    m_pl_ctrl.enqueue_disable_aie_cores();
+
+    m_pl_ctrl.enqueue_halt();
+    m_pl_ctrl.print_micro_codes();
+
+    int ret;
+    bool match = false;
+    int mem_size = 0;
+
+    // XRT auto get group_id
+    auto sender_receiver_k1 =
+        xrt::kernel(device, uuid, "sender_receiver:{sender_receiver_1}");
+    auto controller_k1 =
+        xrt::kernel(device, uuid, "pl_controller_top:{controller_1}");
+
+    // output memory
+    mem_size = num_sample * num_iter * sizeof(int);
```

Comment:
Do you really wish to use sizeof(int)?  I ask for other portions of this method uses sizeof(uint32_t)